docs/utils/getPackages.ts

```diff
@@ -0,0 +1,123 @@
+import fs from 'fs';
+import Task from 'taskarian';
+import { get, HttpError, toHttpTask } from 'ajaxios';
+import glob from 'glob';
+import Decoder, { field, string, succeed } from 'jsonous';
+import path from 'path';
+import { cwd } from 'process';
+import { requireDecoderDuringBuild } from './requireDecoderDuringBuild';
+
+type PackageDir = `${string}/packages/${string}`;
+
+export interface PackageMetadata {
+  name: string;
+  description: string;
+}
+
+export interface PackageData {
+  slug: string;
+  metadata: PackageMetadata;
+  readme: string;
+}
+
+const packageMetadataDecoder: Decoder<PackageMetadata> = succeed({})
+  .assign('name', field('name', string))
+  .assign('description', field('description', string));
+
+const requirePackageMetadataDuringBuild = requireDecoderDuringBuild(packageMetadataDecoder);
+
+const localPackageDirFromSlug = (slug: string): PackageDir => `${cwd()}/../packages/${slug}`;
+
+const getLocalPackageDirs = (): PackageDir[] =>
+  glob.sync(localPackageDirFromSlug('*')) as PackageDir[];
+
+const slugFromPackageDir = (packageDir: PackageDir): string => path.basename(packageDir);
+
+const getLocalPackageSlugs = (): string[] => getLocalPackageDirs().map(slugFromPackageDir);
+
+const localPackageMetadataFromSlug = (slug: string): PackageMetadata =>
+  requirePackageMetadataDuringBuild(
+    JSON.parse(
+      fs.readFileSync(`${localPackageDirFromSlug(slug)}/package.json`, { encoding: 'utf-8' })
```

Comment:
Filesystem interaction should probably be a Task. See `exec_online/.github/actions/lib/packages/filesystem` for some existing implementations which may become another CooperTS package in the future.